docs/proposals/going-towards-to-crd.md

```diff
@@ -0,0 +1,109 @@
+
+- controller reconcile process 
+
+  |                                   | Aggregated API Server               | CRDs                                                                                      |
+  |-----------------------------------|-------------------------------------|-------------------------------------------------------------------------------------------|
+  | Queries                           | uses FieldSelector                  | use LabelSelector, cause the CRD does not support queries via Fields                      |
+  | Removing finalizes                | in `UpdateStatus` method            | in `Update` method                                                                        |
+  | ServiceInstance references fields | via custom `reference` sub-resource | do not support the generic sub-resources, setting them directly via `Update` method |
+
+- `svcat` CLI 
+
+  |         | Aggregated API Server | CRDs              |
+  |---------|--------------------|-------------------|
+  | Queries | uses FieldSelector | use LabelSelector |
+  
+- unit tests coverage - we had to also adjust unit tests because we have a slightly different approach, as you can see above. If we want to have one code base, we need to double tests or support both flows in each test.
+
+- validation of incoming Custom Resources (CRs)
+
+  | Aggregated API Server                                 | CRDs              |
+  |-------------------------------------------------------|-------------------|
+  | `ValidateUpdate` methods and some validation in plugins | Unified and performed only via **ValidatingWebhookConfiguration** |
+
+- defaulting fields of incoming CRs
+
+  | Aggregated API Server                                               | CRDs                                                           |
+  |---------------------------------------------------------------------|----------------------------------------------------------------|
+  | `PrepareForUpdate` methods and default schemas via `defaulter-gen` | Unified and performed only via **MutatingWebhookConfiguration** |
+
+- defining services in Helm charts - different RBACs, secrets, deployments, services, etc.
+
+- underlying constraint of different Kubernetes versions
+
+As you can see above, the Aggregated API Sever use the validation and mutation in a different way. Sometimes the same logic is split across different pkgs. In CRDs, we unified that and copied it directly to the webhook domain. If we want to support both the API and CRDs concepts, we need to invest some time to extract it to some common libraries. Then, in both places use it with an overhead of adjusting the generic interfaces to a custom one. Additionally, after removing the API Server, we need to migrate them back to the domain as having it extracted will only mess the code and additional abstraction layer can confuse other developers.  
```

Comment:
Too many "it-s" and "them-s" -I got lost after a few sentences 😞
Could you paraphrase this paragraph?